spec/requests/list_controller_spec.rb

```diff
@@ -1410,4 +1410,135 @@
       end
     end
   end
+
+  describe "#new" do
+    def extract_topic_ids(response)
+      response.parsed_body["topic_list"]["topics"].map { |topics| topics["id"] }
+    end
+
+    context "when the user is part of the `experimental_new_new_view_groups` site setting group" do
+      fab!(:category) { Fabricate(:category) }
+      fab!(:tag) { Fabricate(:tag) }
+
+      fab!(:new_reply) { Fabricate(:post).topic }
+      fab!(:new_topic) { Fabricate(:post).topic }
+      fab!(:old_topic) { Fabricate(:post).topic }
+
+      fab!(:new_reply_in_category) do
+        Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+      end
+      fab!(:new_topic_in_category) do
+        Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+      end
+      fab!(:old_topic_in_category) do
+        Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+      end
+
+      fab!(:new_reply_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+      fab!(:new_topic_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+      fab!(:old_topic_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+
+      before do
+        SiteSetting.experimental_new_new_view_groups = group.name
+        group.add(user)
+        sign_in(user)
+
+        [topic, old_topic, old_topic_in_category, old_topic_with_tag].each do |topic|
+          TopicUser.update_last_read(user, topic.id, 1, 1, 1)
+        end
+
+        [new_reply, new_reply_in_category, new_reply_with_tag].each do |topic|
+          TopicUser.change(
+            user.id,
+            topic.id,
+            notification_level: TopicUser.notification_levels[:tracking],
+          )
+          TopicUser.update_last_read(user, topic.id, 1, 1, 1)
+          Fabricate(:post, topic: topic)
+        end
```

Comment:
I think these can be done during fabrication instead.

Reply:
done!